nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/marshallers/json/document/DocumentPropertyJsonWriter.java

```diff
@@ -92,14 +92,12 @@ protected void writeProperty(JsonGenerator jg, Property prop) throws IOException
             writeScalarProperty(jg, prop);
         } else if (prop.isList()) {
             writeListProperty(jg, prop);
-        } else {
-            if (prop.isPhantom()) {
-                jg.writeNull();
-            } else if (prop instanceof BlobProperty) { // a blob
-                writeBlobProperty(jg, prop);
-            } else { // a complex property
-                writeComplexProperty(jg, prop);
-            }
+        } else if (prop.isComplex()) {
+            writeComplexProperty(jg, prop);
+        } else if (prop instanceof BlobProperty) { // a blob
```

Comment:
Because a `BlobProperty` is also a complex property, this check should come before `prop.isComplex()`. In my local tests I got a `NuxeoException` with `Trying to access a member of a null object: mime-type` when trying to write a blob property, which was treated as a complex.